keyboards/edinburgh41/keymaps/lalit/keymap.c

```diff
@@ -0,0 +1,111 @@
+/* Copyright 2022 schwarzer-geiger
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
+#include "thumbstick.c"
```

Comment:
```suggestion
```
Do not include .c files. Make a header and add the .c file to SRC, or put it all in keymap.c.

Reply:
Dear fauxpark,
what do you mean exactly by making a header? What I currently did is define all the thumbstick parameters in the config.h, include the config.h in the thumbstick.c file and added thumbstick.c to SRC in rules.mk. Unfortunately, the functionality defined in thumbstick.c is not performed. What should I do exactly?

Many thanks,
Lalit

Reply:
The problem with including `.c` files is that if you include it in multiple other `.c` files, you will get multiple copies of the same things declared in it. Standard practice is to create a `.h` file that simply declares the function prototypes, while the `.c` file contains the implementations. All the keymap.c needs to know is that those symbols exist, and the compiler and linker are smart enough to pull it all together.

Reply:
Dear reviewers,
sadly I'm still struggling with the importing and I was wondering whether you could help me out again. I updated my edinburgh41 repository in my dev_branch to reflect my current version. The structure is currently:
Rules.mk has thumbstick.c added to SRC.
thumbstick.c imports config.h and post_config.h which define parameters used in thumbstick.c.
In my understanding thumbstick.c gets compiled and has access to config.h, so everything should work. However, what actually happens at the moment is:
The firmware compiles without errors, but the thumbstick simply doesn't respond to actions that rely on the parameters defined in config.h. I had this all working in the first PR, but the import structure wasn't to QMK standards, so now I'm trying to get it working and to standard. So I'd be really grateful if somebody could have a quick look and see how I can get this working. Please let me know if you need any more info or explanation.
Many thanks!

Reply:
For importing C filse, you want to add `SRC += filename.c` to the rules.mk.  This adds it to the list of compile files.